Network/AMQP/Internal.hs

```diff
@@ -486,6 +486,14 @@ data Channel = Channel {
                     chanExceptionHandlers :: MVar [CE.SomeException -> IO ()]
                 }
 
+data ChanThreadKilledException = ChanThreadKilledException { cause :: CE.SomeException }
```

Comment:
You can keep the definition here and export it from `Network.AMQP`. `Network.AMQP.Types` is mostly for AMQP-defined data-types (I'm not completely sure why I put AMQPException there).